packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -116,153 +190,393 @@ class TimeInput extends React.Component {
+      const minute = type === TimeUtil.inputType.MINUTE ? timeValue : this.state.minute;
+
+      if (hour === '' && minute === '') {
+        this.props.onChange(event, '');
       } else {
-        changedValue = this.state.hour.concat(':', timeValue);
+        this.props.onChange(event, this.formatHour(hour, meridiem).concat(':', minute));
+      }
+    }
+  }
+
+  formatHour(hour, meridiem) {
+    if (!hour) {
+      return hour;
+    }
+
+    let tempHour = parseInt(hour, 10);
+
+    if (this.props.variant === TimeUtil.FORMAT_12_HOUR) {
+      if (meridiem === this.postMeridiem && tempHour < 12) {
+        tempHour += 12;
+      } else if (meridiem === this.anteMeridiem && tempHour === 12) {
+        tempHour = 0;
       }
+    }
+
+    let hourString = tempHour.toString();
 
-      this.props.onChange(event, changedValue);
+    if (hourString.length < 2) {
+      hourString = '0'.concat(hourString);
     }
+
+    return hourString;
   }
 
-  render() {
+  handleMeridiemInputKeyDown(event) {
+    if (event.keyCode === TimeUtil.keyCodes.ARROWLEFT ||
+        event.keyCode === TimeUtil.keyCodes.DELETE ||
+        event.keyCode === TimeUtil.keyCodes.BACKSPACE) {
+      this.minuteInput.textInput.focus();
+      event.preventDefault();
+    }
+  }
+
+  mobileInput() {
     const {
       inputAttributes,
+      minuteAttributes,
+      hourAttributes,
       onBlur,
       onChange,
       name,
       value,
+      variant,
+      ...customProps
+    } = this.props;
+
+    // Using the state of hour and minute create a time in UTC represented in ISO 8601 format.
+    let timeValue = '';
+
+    if (this.state.hour.length > 0 || this.state.minute.length > 0) {
+      let hour = parseInt(this.state.hour, 10);
+
+      if (this.props.variant === TimeUtil.FORMAT_12_HOUR && this.state.meridiem === this.postMeridiem && hour > 12) {
+        hour += 12;
+      }
+
+      timeValue = 'T'.concat(hour, ':', this.state.minute);
+    }
+
+    if (hourAttributes && !hourAttributes.id) {
+      hourAttributes.id = 'terra-time-hour-'.concat(name);
+    }
+
+    if (minuteAttributes && !minuteAttributes.id) {
+      minuteAttributes.id = 'terra-time-minute-'.concat(name);
+    }
+
+    return (
+      <div {...customProps} className={cx(['mobile-time-picker', customProps.className])}>
+        <input
+          // Create a hidden input for storing the name and value attributes to use when submitting the form.
+          // The data stored in the value attribute will be the visible date in the date input but in ISO 8601 format.
+          type="hidden"
+          name={name}
+          value={timeValue}
+        />
+        <div className={cx('time-input-group')}>
+          <Input
+            {...inputAttributes}
+            {...hourAttributes}
+            ref={(inputRef) => { this.hourInput = inputRef; }}
+            className={cx('time-input-hour')}
+            type="tel"
```

Comment:
What is the reason for making this a tel input type instead of text?

Reply:
For mobile devices, UX wants to use just the numeric pad.

Reply:
Does setting the pattern prop on lines 517 and 540 not work? This prop was added by this [change](https://github.com/cerner/terra-core/pull/365/commits/336eb59538fd24e378d3041bd35062d8a6a41921) to use the numeric keyboard on mobile devices. I think the pattern prop can be removed if setting the type to tel also works.

Reply:
The pattern does do the trick. I do see that tel gives us additional functionality on the keyboard, but I don't see the need for +, * and # values. I'll sync with UX on what they like.

Reply:
I decided to remove the tel for now. It gives us the control we need, and I don't see the need for those other inputs that appear. https://github.com/cerner/terra-core/pull/1027/commits/6369b5318e8d2e8b3deaae1332d0ff0369498f90